graph/src/data/query/query.rs

```diff
@@ -112,19 +112,20 @@ impl serde::ser::Serialize for QueryVariables {
 
 #[derive(Clone, Debug)]
 pub enum QueryTarget {
-    Name(SubgraphName),
-    Deployment(DeploymentHash),
+    Name(SubgraphName, VersionNumber),
+    Deployment(DeploymentHash, VersionNumber),
 }
 
-impl From<DeploymentHash> for QueryTarget {
-    fn from(id: DeploymentHash) -> Self {
-        Self::Deployment(id)
+impl QueryTarget {
+    pub fn get_version_number(&self) -> VersionNumber {
+        match self {
+            Self::Name(_, version) => version.clone(),
+            Self::Deployment(_, version) => version.clone(),
+        }
     }
-}
```

Comment:
I tried to look for the best place to put the requested version and `QueryTarget` won the contest.